src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyCaseBlocksWithSwitchRuleAloneOrSingleline.java

```diff
@@ -0,0 +1,104 @@
+/*
+RightCurly
+option = ALONE_OR_SINGLELINE
+tokens = LITERAL_CASE
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyCaseBlocksWithSwitchRuleAloneOrSingleline {
+
+    public void testt0() {
+        int mode = 0;
+        switch (mode) {
+            case 1 -> {int x = 1;}   // ok, single line
+            case 2 -> {int x = 0;}
+            case 3 -> System.out.println("x is 3");
+        }
+    }
+
+    public static void test0() {
+        int mode = 0;
+        switch (mode) {
+            case 1 -> {
+                int
+            x = 1;} // violation '}' at column 19 should be alone on a line'
+            case 2 -> {
+                int x = 0;
+            }
+            case 3 -> System.out.println("x is 3");
+        }
+    }
+
+    public static void test() {
+        int mode = 0;
+        switch (mode) {
+            case 1 -> {
+                int x = 1;
+
+            } default -> {          // violation '}' at column 13 should be alone on a line'
+                int x = 0;
+            }
+        }
+    }
+
+    public static void test1() {
+        int k = 0;
+        switch (k) {
+            case 1 -> {
+
+
+           int x = 1; } case 2 -> {       // violation '}' at column 23 should be alone on a line'
+                int x = 2;}     // violation '}' at column 27 should be alone on a line'
+
+        }
+    }
+
+    public static void test2() {
+         int k = 0;
+         switch (k) {
+            case 1 -> {int x = 1;
+
+            }case 2 -> throw new RuntimeException();
+            // violation above '}' at column 13 should be alone on a line'
+         }
+    }
+
+    public static void test3() {
+         int k = 0;
+         switch (k) {
+            case 1 -> {
+                 int x = 1;
+            }
+            case 2 -> {
+                    int x = 2;
+            } }      // violation '}' at column 13 should be alone on a line'
+    }
+
+    public static void test4() {
+        int mode = 0;
+        switch (mode) { case 0 -> {int x = 1;} default -> {int x = 0;} }
```

Comment:
Continuing from https://github.com/checkstyle/checkstyle/pull/14623#discussion_r1555180807, shouldn't this be a violation since it doesn't end the line?

Reply:
the thing is both `LITERAL_CASE` and `LITERAL_DEFAULT` are both under `SWITCH_RULE` token
```
            |   |--SWITCH_RULE -> SWITCH_RULE [7:10]     <----
            |   |   |--LITERAL_CASE -> case [7:10]
            |   |   |   `--EXPR -> EXPR [7:15]
            |   |   |       `--NUM_INT -> 0 [7:15]
            |   |   |--LAMBDA -> -> [7:17]
            |   |   `--SLIST -> { [7:20]
            |   |       `--RCURLY -> } [7:21]
            |   |--SWITCH_RULE -> SWITCH_RULE [7:23]      <-----
            |   |   |--LITERAL_DEFAULT -> default [7:23]
            |   |   |--LAMBDA -> -> [7:31]
            |   |   `--SLIST -> { [7:34]
            |   |       `--RCURLY -> } [7:35]

```
So, this example is the same as  ` switch (mode) { case 0 -> {int x = 1;} case 1 -> {int x = 0;} }`  and this is ok because it is a single line format of a block in this example `}` shouldn't be at the end of the line.

This should be violation if the nextToken is not `SWITCH_RULE` example  `switch (mode) { case 0 -> {int x = 1;} int y;`
here will be a violation and `}` should be at the end of the line

Reply:
Yes I was missing that we should not consider this as a multiblock statement So it should be violation
Done